src/LZ4Format.cpp

```diff
@@ -62,14 +62,14 @@ std::string lz4::DecompressLZ4(const char *data, size_t length)
 	return out;
 }
 
-std::unique_ptr<char> lz4::CompressLZ4(const std::string &data, const int lz4_preset, std::size_t &outSize)
+std::unique_ptr<char[]> lz4::CompressLZ4(const std::string &data, const int lz4_preset, std::size_t &outSize)
 {
 	PROFILE_SCOPED()
 	LZ4F_preferences_t pref = LZ4F_INIT_PREFERENCES;
 	pref.compressionLevel = lz4_preset;
 
 	std::size_t compressBound = LZ4F_compressFrameBound(data.size(), &pref);
-	std::unique_ptr<char> out(new char[compressBound]);
+	std::unique_ptr<char[]> out(new char[compressBound]);
```

Comment:
...ah, I've been writing too much C. I forgot that `char[]*` and a `char *` to some length of `char`s are two different types.